Gems/Atom/RPI/Code/Source/RPI.Public/Pass/CopyPass.cpp

```diff
@@ -84,6 +92,70 @@ namespace AZ
+                AZ_Assert(
+                    device1->GetFeatures().m_signalFenceFromCPU,
+                    "CopyPass: Device to device copy is only possible if all devices support signalling fences from the CPU");
+                [[maybe_unused]] auto* device2 =
+                    RHI::RHISystemInterface::Get()->GetDevice(m_data.m_destinationDeviceIndex != RHI::MultiDevice::InvalidDeviceIndex ? m_data.m_destinationDeviceIndex : RHI::MultiDevice::DefaultDeviceIndex);
+                AZ_Assert(
+                    device2->GetFeatures().m_signalFenceFromCPU,
+                    "CopyPass: Device to device copy is only possible if all devices support signalling fences from the CPU");
+
+                for (auto& fence : m_device1SignalFence)
+                {
+                    fence = new RHI::MultiDeviceFence();
+                    AZ_Assert(fence != nullptr, "CopyPass failed to create a fence");
+                    [[maybe_unused]] RHI::ResultCode result = fence->Init(RHI::MultiDevice::AllDevices, RHI::FenceState::Signaled);
+                    AZ_Assert(result == RHI::ResultCode::Success, "CopyPass failed to init fence");
+                }
+
+                for (auto& fence : m_device2WaitFence)
+                {
+                    fence = new RHI::MultiDeviceFence();
+                    AZ_Assert(fence != nullptr, "CopyPass failed to create a fence");
+                    [[maybe_unused]] auto result = fence->Init(RHI::MultiDevice::AllDevices, RHI::FenceState::Signaled);
+                    AZ_Assert(result == RHI::ResultCode::Success, "CopyPass failed to init fence");
+                }
+
+                m_copyScopeProducerDeviceToHost = AZStd::make_shared<RHI::ScopeProducerFunctionNoData>(
+                    RHI::ScopeId{ AZStd::string(GetPathName().GetStringView()) },
+                    AZStd::bind(&CopyPass::SetupFrameGraphDependenciesDeviceToHost, this, AZStd::placeholders::_1),
+                    AZStd::bind(&CopyPass::CompileResourcesDeviceToHost, this, AZStd::placeholders::_1),
+                    AZStd::bind(&CopyPass::BuildCommandListInternalDeviceToHost, this, AZStd::placeholders::_1),
+                    m_hardwareQueueClass,
+                    m_data.m_sourceDeviceIndex);
+
+                m_copyScopeProducerHostToDevice = AZStd::make_shared<RHI::ScopeProducerFunctionNoData>(
+                    RHI::ScopeId{ AZStd::string(GetPathName().GetStringView()) + "_2" },
+                    AZStd::bind(&CopyPass::SetupFrameGraphDependenciesHostToDevice, this, AZStd::placeholders::_1),
+                    AZStd::bind(&CopyPass::CompileResourcesHostToDevice, this, AZStd::placeholders::_1),
+                    AZStd::bind(&CopyPass::BuildCommandListInternalHostToDevice, this, AZStd::placeholders::_1),
+                    m_hardwareQueueClass,
+                    m_data.m_destinationDeviceIndex);
+            }
+            //@TODO scope queries (not sure how to do this for two different devices)
```

Comment:
Does this comment need to be here? It doesn't seem to provide any additional information.